README.md

```diff
@@ -95,6 +95,41 @@ $ sudo cat /sys/kernel/debug/tracing/trace_pipe
            <...>-461101 [018] d... 505434.345367: bpf_trace_printk: UPROBE EXIT: return = 5
 ```
 
+# Kprobe
+
+`kprobe` is an example of dealing with kernel-space entry and exit (return) probes,
+`kprobe` and `kretprobe` in libbpf lingo. It attaches `kprobe` and `kretprobe`
+BPF programs to the `kernel_clone` function and logs some input arguments
+and return result, respectively, using `bpf_printk()` macro. The example is similar to the
+Linux kernel's `samples/kprobe/kprobe_example.c` example. The example will catch calls to
+`fork`, `clone` and similar functions. Depending on the function called different
+arguments will be set. See the corresponding man pages for more details.
```

Comment:
It would indeed be good to call out fentry/fexit as a nicer alternative to kprobe/kretprobe (in terms of usability and performance). Adding fentry demo, that matches kprobe demo would allow to contrast them to each other. So how about doing just that?

Reply:
I've added a fentry exec() example. However, I'm not sure if it will be the best example due to a few issues. It also attaches to `bprm_execve` so that it matches the kprobe example and to show the ability to use "normal" C in fentry/fexit programs as compared to kprobes. I originally tried to attach to `do_execveat_common`. The alternative was to attach to architecture specific execve syscalls, like `__x64_sys_execveat` which I don't think would make sense for an example. 

`do_execveat_common` worked fine with kprobes, but then I encountered the first issue with fentry. Since libbpf seems to currently use BTF to attach fentry/fexit programs, it couldn't attach to `do_execveat_common` on my system which is running latest Arch Linux kernel. 
The reason for that seems to be that `do_execveat_common` is a static kernel function, and my guess is that the kernel BTF wasn't generated with a recent enough pahole. After manually compiling a 5.10 kernel from source the BTF generated that way included `do_execveat_common`. I assume it works because `do_execveat_common` is listed in the ftrace function list, and newer pahole versions seem to check for that. The error message from libbpf for the vanilla Arch kernel is below:
```
libbpf: do_execveat_common is not found in vmlinux BTF
```

That raises the second issue, I still couldn't attach to `do_execveat_common` with the new kernel because of the following error:
```
The function do_execveat_common arg2 type STRUCT is unsupported.
processed 0 insns (limit 1000000) max_states_per_insn 0 total_states 0 peak_states 0 mark_read 0
```

From what I understand, BTF doesn't support plain struct function arguments, only pointers to structs and `do_execveat_common` has two `struct user_arg_ptr` arguments. Due to all that I decided to hook  `bprm_execve` which does work with fentry/fexit. It will still need kernel BTF generated with recent enough pahole because `bprm_execve` is also a static function in `fs/exec.c`.

At this point I'm not sure whether it would be better to attach to another function, or document the requirements to run this example.